bindings/c/include/opendal.h

```diff
@@ -25,14 +25,229 @@
 #include <stddef.h>
 #include <stdbool.h>
 
+/*
+ The C-compatible error type enum used in c bindings.
+ NOTICE: the type definition of [`ErrorNo`] should be aligned with
+         [`od::ErrorKind`]
+ */
+typedef enum opendal_errno {
```

Comment:
Adding a new concept seems not a good idea, how about keeping `opendal_error_kind`?

Reply:
Of course we can keep it 😄!

I use this only because errno is a convention used in C.